libs/x-content/impl/src/main/java/org/elasticsearch/xcontent/provider/json/JsonXContentImpl.java

```diff
@@ -96,21 +98,37 @@ private XContentParser createParser(XContentParserConfiguration config, JsonPars
 
     @Override
     public XContentParser createParser(XContentParserConfiguration config, String content) throws IOException {
-        return createParser(config, jsonFactory.createParser(content));
+        try {
+            return createParser(config, jsonFactory.createParser(content));
+        } catch (CharConversionException e) {
+            throw new XContentParseException(null, e.getMessage(), e);
```

Comment:
JsonFactory.createParser can throw JsonParseException, so shouldn't this catch also use handleParseException?